compiler/noirc_evaluator/src/ssa/opt/loop_invariant.rs

```diff
@@ -697,4 +748,53 @@ mod test {
         let ssa = ssa.loop_invariant_code_motion();
         assert_normalized_ssa_equals(ssa, expected);
     }
+
+    #[test]
+    fn transform_safe_ops_to_unchecked_during_code_motion() {
+        // This test is identical to `simple_loop_invariant_code_motion`, except this test
+        // uses a checked add in `b3`.
+        let src = "
+        brillig(inline) fn main f0 {
+          b0(v0: i32, v1: i32):
+              jmp b1(i32 0)
+          b1(v2: i32):
+              v5 = lt v2, i32 4
+              jmpif v5 then: b3, else: b2
+          b2():
+              return
+          b3():
+              v6 = mul v0, v1
+              constrain v6 == i32 6
+              v8 = add v2, i32 1
+              jmp b1(v8)
+        }
+        ";
+
+        let ssa = Ssa::from_str(src).unwrap();
+        let main = ssa.main();
+
+        let instructions = main.dfg[main.entry_block()].instructions();
+        assert_eq!(instructions.len(), 0); // The final return is not counted
+
+        // `v8 = add v2, i32 1` in b3 should now be `v9 = unchecked_add v2, i32 1` in b3
```

Comment:
The SSA parser supports comments (and trailing comments) so, if it's clearer, you could put this comment right next to the relevant line (or not! up to you)

Reply:
I preferred the comment here as it is colored with green text rather than the same text of the SSA.